### DISCUSSION TITLE
no alerts and logs in hunt from idh

### DISCUSSION THREAD

**Discussion by Taraskobilskiy** [Security-Onion-Solutions/securityonion#12512] [Unsupported Versions] [answered]
### Version

2.4.50

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

32

### RAM

64

### Storage for /

500

### Storage for /nsm

500

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

when installing the idh module and deploying it to the manager node, there are no logs from opencanary, playbooks are active, http bait is available, but nothing happens when entering the login and password. I tried to do everything that was described in the community but there was no result. 

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/156078629/84e8da40-d706-4e8a-9e70-a0011b245e3f)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/156078629/fc810264-78db-456f-a6bc-1d2d138a1f06)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/156078629/351e5fd4-1fc4-490b-88ad-3382691b808d)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/156078629/206155c1-99c8-4020-8b6b-64f0a348598e)

### Guidelines

- [x] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by foxman0719** [comment]
I have the same problem, I do TCPDUMP to the interfaces and there is traffic without problem, but it does not appear reflected in the tool or in alerts or in the GRID

please help

**Comment by Taraskobilskiy** [comment]
UPD: logs 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/156078629/354de5ed-5412-49ba-a1e2-cb3b9cf925d4)

**Comment by Taraskobilskiy** [comment]
The problem still persists, could you please suggest steps to troubleshoot since it’s not clear what to do about it

**Comment by Taraskobilskiy** [comment]
I’ll add: when installing node idh, the installation occurs correctly, the node itself is visible in the grid, there are no errors, but when accessing in IDH port 22 ssh or 80 http and entering login and password, opencanary events are not generated (at least they are not visible in hunt)

**Comment by reyesj2** [comment]
Do you see logs generated in ```/nsm/idh/opencanary.log``` on the idh node?
What is the output of ```elastic-agent status``` on the idh node?

**Reply by Taraskobilskiy** [reply]
Looks like the problem is with the agent

elastic-agent status:
***
┌─ fleet
│  └─ status: (HEALTHY) Connected
└─ elastic-agent
   ├─ status: (DEGRADED) 1 or more components/units in a degraded state
   ├─ filestream-monitoring
   │  ├─ status: (DEGRADED) Degraded: pid '95258' missed 1 check-in
   │  ├─ filestream-monitoring
   │  │  └─ status: (STARTING) Starting: spawned pid '95258'
   │  └─ filestream-monitoring-filestream-monitoring-agent
   │     └─ status: (STARTING) Starting: spawned pid '95258'
   ├─ log-default
   │  ├─ status: (DEGRADED) Degraded: pid '95219' missed 1 check-in
   │  ├─ log-default
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-elasticsearch-90a626e3-36d8-4399-a9b9-39361f7448da
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-logs-15b78604-37fb-46cc-9cd4-cd4692f53122
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-logs-1b2c77b7-36bf-4212-88b1-895ffc5afdf1
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-logs-1fd4c726-bc9d-4c21-9f84-3e78a519f2a8
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-logs-5cc786b7-a311-4f5a-9738-dbad9b1ab643
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-logs-689c5fcf-d70d-445a-ac71-fb5a407d04e1
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-logs-81d11465-6f80-43e1-b289-42678e41142e
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-logs-8a6b5755-62e8-4fab-89aa-fd7a9916ff8b
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-logs-9e01c157-1694-4cdf-abfa-dd4682ce401e
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-logs-a646f5db-391f-429c-857d-948eb6a781b2
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-logs-db6fed93-57de-4e4d-99f5-4192207f2b2f
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-logs-dea3e095-79b6-477f-9a59-ad7334a1016e
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-logs-e56ad89f-5856-4eb4-ace1-43177c8957a1
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-logs-zeek-logs
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  ├─ log-default-logfile-redis-9b473127-9a2e-4628-86a9-2f8579a93d05
   │  │  └─ status: (STARTING) Starting: spawned pid '95219'
   │  └─ log-default-logfile-system-21d7b3bb-c260-45d5-b736-f6b47741e224
   │     └─ status: (STARTING) Starting: spawned pid '95219'
   ├─ osquery-default
   │  ├─ status: (DEGRADED) Degraded: pid '95217' missed 1 check-in
   │  ├─ osquery-default
   │  │  └─ status: (STARTING) Starting: spawned pid '95217'
   │  └─ osquery-default-727f3426-a6ec-446d-8e40-104f1cddaaad
   │     └─ status: (STARTING) Starting: spawned pid '95217'
   ├─ tcp-default
   │  ├─ status: (DEGRADED) Degraded: pid '95218' missed 1 check-in
   │  ├─ tcp-default
   │  │  └─ status: (STARTING) Starting: spawned pid '95218'
   │  └─ tcp-default-tcp-tcp-8ec50926-73e4-463f-a048-ae94abb68b75
   │     └─ status: (STARTING) Starting: spawned pid '95218'
   └─ udp-default
      ├─ status: (DEGRADED) Degraded: pid '95216' missed 1 check-in
      ├─ udp-default
      │  └─ status: (STARTING) Starting: spawned pid '95216'
      └─ udp-default-udp-udp-c4e5e9fc-9b1c-4166-b3bd-6fdf96528b08
         └─ status: (STARTING) Starting: spawned pid '95216'
***
yes, the log contains data about authorization attempts
/nsm/idh/opencanary.log:
***
......
{"dst_host": "10.189.96.26", "dst_port": 22, "local_time": "2024-03-13 13:23:05.189662", "local_time_adjusted": "2024-03-13 13:23:05.189692", "logdata": {"LOCALVERSION": "SSH-2.0-OpenSSH_5.1p1 Debian-4", "PASSWORD": "TESTso123", "REMOTEVERSION": "SSH-2.0-OpenSSH_for_Windows_8.6", "USERNAME": "testSO"}, "logtype": 4002, "node_id": "so-idh4", "src_host": "192.168.32.55", "src_port": 57636, "utc_time": "2024-03-13 13:23:05.189687"}
***

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/156078629/3dd2a47a-8570-4baf-8505-5f42bad2509d)


please tell me what could happen in elastic agent?

**Reply by Taraskobilskiy** [reply]
Maybe I have too many integrations and the extra ones need to be removed from the policy?

**Reply by reyesj2** [reply]
No that looks correct, the IDH node uses the so-grid-nodes_general policy. The 'idh-logs' integration is what will pick up the logs if found.

You can click 'logs' in that screenshot and see if you can find what the error causing the unhealthy state is.

Is this your only idh node? If it is just go to hunt and search ```tags: idh``` you'll see all idh logs. If you have more than one idh node you'll need to look for the ```node_id``` and make sure 'so-idh4' is included in the logs shown

You can also try reinstalling the elastic agent on that idh node
For that from your manager run
``` 
sudo salt 'so-idh4*' cmd.run 'elastic-agent uninstall -f'
sudo salt 'so-idh4*' state.apply elasticfleet.install_agent_grid
sudo salt 'so-idh4*' cmd.run 'elastic-agent status'

```

**Reply by Taraskobilskiy** [reply]
The logs are also not displayed, I tried to run diagnostics and downloaded an archive with logs, but I didn’t find anything there to help solve the problem:

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/156078629/534b470f-bac1-41ae-b07e-754ef69f40bb)


I deployed more than 5 IDH, but in the end not one of them works. Searching by tag also does not give results, 
***
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/156078629/8574bbcb-ac9a-469e-9adf-018186ebe5aa)
***
although if you take a very long period, you can see that at first there were events: 
***
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/156078629/def218ef-5a73-4fe0-8f3a-05c11e89730e)
***

**Reply by Taraskobilskiy** [reply]
UPD:
after reinstalling the agent everything looks like this, but there are still no events:
***

[sysadmin@so-manager ~]$ sudo salt 'so-idh4*' cmd.run 'elastic-agent uninstall -f'
[sudo] password for sysadmin:
so-idh4_idh:
    Elastic Agent has been uninstalled.
[sysadmin@so-manager ~]$ sudo salt 'so-idh4*' state.apply elasticfleet.install_agent_grid
so-idh4_idh:
----------
          ID: run_installer
    Function: cmd.script
        Name: salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64
      Result: True
     Comment: Command 'salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64' run
     Started: 04:25:59.639441
    Duration: 13953.109 ms
     Changes:
              ----------
              pid:
                  315458
              retcode:
                  0
              stderr:
              stdout:

                  Installation initiated, view install log for further details.


                  Installation completed successfully.

                  Summary for so-idh4_idh
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:  13.953 s

[sysadmin@so-manager ~]$ sudo salt 'so-idh4*' cmd.run 'elastic-agent status'
so-idh4_idh:
    ┌─ fleet
    │  └─ status: (HEALTHY) Connected
    └─ elastic-agent
       ├─ status: (HEALTHY) Running
       ├─ filestream-monitoring
       │  ├─ status: (STARTING) Starting: spawned pid '318133'
       │  ├─ filestream-monitoring
       │  │  └─ status: (STARTING) Starting: spawned pid '318133'
       │  └─ filestream-monitoring-filestream-monitoring-agent
       │     └─ status: (STARTING) Starting: spawned pid '318133'
       ├─ log-default
       │  ├─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-elasticsearch-90a626e3-36d8-4399-a9b9-39361f7448da
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-logs-15b78604-37fb-46cc-9cd4-cd4692f53122
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-logs-1b2c77b7-36bf-4212-88b1-895ffc5afdf1
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-logs-1fd4c726-bc9d-4c21-9f84-3e78a519f2a8
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-logs-5cc786b7-a311-4f5a-9738-dbad9b1ab643
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-logs-689c5fcf-d70d-445a-ac71-fb5a407d04e1
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-logs-81d11465-6f80-43e1-b289-42678e41142e
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-logs-8a6b5755-62e8-4fab-89aa-fd7a9916ff8b
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-logs-9e01c157-1694-4cdf-abfa-dd4682ce401e
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-logs-a646f5db-391f-429c-857d-948eb6a781b2
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-logs-db6fed93-57de-4e4d-99f5-4192207f2b2f
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-logs-dea3e095-79b6-477f-9a59-ad7334a1016e
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-logs-e56ad89f-5856-4eb4-ace1-43177c8957a1
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-logs-zeek-logs
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  ├─ log-default-logfile-redis-9b473127-9a2e-4628-86a9-2f8579a93d05
       │  │  └─ status: (STARTING) Starting: spawned pid '318118'
       │  └─ log-default-logfile-system-21d7b3bb-c260-45d5-b736-f6b47741e224
       │     └─ status: (STARTING) Starting: spawned pid '318118'
       ├─ osquery-default
       │  ├─ status: (STARTING) Starting: spawned pid '318104'
       │  ├─ osquery-default
       │  │  └─ status: (STARTING) Starting: spawned pid '318104'
       │  └─ osquery-default-727f3426-a6ec-446d-8e40-104f1cddaaad
       │     └─ status: (STARTING) Starting: spawned pid '318104'
       ├─ tcp-default
       │  ├─ status: (STARTING) Starting: spawned pid '318127'
       │  ├─ tcp-default
       │  │  └─ status: (STARTING) Starting: spawned pid '318127'
       │  └─ tcp-default-tcp-tcp-8ec50926-73e4-463f-a048-ae94abb68b75
       │     └─ status: (STARTING) Starting: spawned pid '318127'
       └─ udp-default
          ├─ status: (STARTING) Starting: spawned pid '318097'
          ├─ udp-default
          │  └─ status: (STARTING) Starting: spawned pid '318097'
          └─ udp-default-udp-udp-c4e5e9fc-9b1c-4166-b3bd-6fdf96528b08
             └─ status: (STARTING) Starting: spawned pid '318097'
***

**Reply by Taraskobilskiy** [reply]
upd:
after some time, some services switched to DEGRADED status and the error "ERROR: Minions returned with non-zero exit code" appeared:

***
[sysadmin@so-manager ~]$ sudo salt 'so-idh4*' cmd.run 'elastic-agent status'
so-idh4_idh:
    ┌─ fleet
    │  └─ status: (HEALTHY) Connected
    └─ elastic-agent
       ├─ status: (DEGRADED) 1 or more components/units in a degraded state
       ├─ filestream-monitoring
       │  ├─ status: (DEGRADED) Degraded: pid '320445' missed 2 check-ins
       │  ├─ filestream-monitoring
       │  │  └─ status: (STARTING) Starting: spawned pid '320445'
       │  └─ filestream-monitoring-filestream-monitoring-agent
       │     └─ status: (STARTING) Starting: spawned pid '320445'
       ├─ log-default
       │  ├─ status: (DEGRADED) Degraded: pid '320433' missed 2 check-ins
       │  ├─ log-default
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-elasticsearch-90a626e3-36d8-4399-a9b9-39361f7448da
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-logs-15b78604-37fb-46cc-9cd4-cd4692f53122
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-logs-1b2c77b7-36bf-4212-88b1-895ffc5afdf1
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-logs-1fd4c726-bc9d-4c21-9f84-3e78a519f2a8
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-logs-5cc786b7-a311-4f5a-9738-dbad9b1ab643
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-logs-689c5fcf-d70d-445a-ac71-fb5a407d04e1
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-logs-81d11465-6f80-43e1-b289-42678e41142e
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-logs-8a6b5755-62e8-4fab-89aa-fd7a9916ff8b
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-logs-9e01c157-1694-4cdf-abfa-dd4682ce401e
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-logs-a646f5db-391f-429c-857d-948eb6a781b2
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-logs-db6fed93-57de-4e4d-99f5-4192207f2b2f
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-logs-dea3e095-79b6-477f-9a59-ad7334a1016e
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-logs-e56ad89f-5856-4eb4-ace1-43177c8957a1
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-logs-zeek-logs
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  ├─ log-default-logfile-redis-9b473127-9a2e-4628-86a9-2f8579a93d05
       │  │  └─ status: (STARTING) Starting: spawned pid '320433'
       │  └─ log-default-logfile-system-21d7b3bb-c260-45d5-b736-f6b47741e224
       │     └─ status: (STARTING) Starting: spawned pid '320433'
       ├─ osquery-default
       │  ├─ status: (DEGRADED) Degraded: pid '320427' missed 2 check-ins
       │  ├─ osquery-default
       │  │  └─ status: (STARTING) Starting: spawned pid '320427'
       │  └─ osquery-default-727f3426-a6ec-446d-8e40-104f1cddaaad
       │     └─ status: (STARTING) Starting: spawned pid '320427'
       ├─ tcp-default
       │  ├─ status: (DEGRADED) Degraded: pid '320439' missed 2 check-ins
       │  ├─ tcp-default
       │  │  └─ status: (STARTING) Starting: spawned pid '320439'
       │  └─ tcp-default-tcp-tcp-8ec50926-73e4-463f-a048-ae94abb68b75
       │     └─ status: (STARTING) Starting: spawned pid '320439'
       └─ udp-default
          ├─ status: (DEGRADED) Degraded: pid '320405' missed 2 check-ins
          ├─ udp-default
          │  └─ status: (STARTING) Starting: spawned pid '320405'
          └─ udp-default-udp-udp-c4e5e9fc-9b1c-4166-b3bd-6fdf96528b08
             └─ status: (STARTING) Starting: spawned pid '320405'
ERROR: Minions returned with non-zero exit code
***

**Comment by Taraskobilskiy** [comment]
Can you suggest any solution to this problem? or other troubleshooting actions?

**Comment by reyesj2** [comment, marked as answer]
From elastic fleet when you click on the 'so-idh4' agent you should be able to pull up logs about that agent and maybe why its degraded. 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/94730068/1db6254f-e24b-453b-8876-dbd81715eaa3)

Click on 'Dataset' and unselect any that are selected so it shows all logs.

The so-idh4 node status in SOC -> Grid is still OK/green?

**Reply by Taraskobilskiy** [reply]
unfortunately, the logs are not displayed for any selection in the dataset:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/156078629/4a09e312-d36c-44e1-afc6-fbd5d3a853b2)

Yes, the grid status is green:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/156078629/1a656562-8bbf-4c32-818f-66dd534703f7)

**Comment by Taraskobilskiy** [comment]
In general, we managed to solve the problem by reinstalling IDH on a separate network and setting up a firewall rule in the admin panel.

but no matter how much I deployed the idh module on the same network where the manager node and other sec onion elements were, the problem with the elastic agent remained.

@reyesj2  thank you for your feedback and help in this matter!
